genai-perf/docs/analyze.md

```diff
@@ -0,0 +1,240 @@
+
+#### `--sweep-type` - The type of stimulus you wish the sweep over
+The currently support stimulus values are `concurrency`, `request_rate`, `input_sequence_length`, and `num_dataset_entries`
+
+#### `--sweep-range` - The range over which the stimulus will be swept
+This can be represented as `min:max` or `min:max:step`. If a `step` is not specified then we assume the range to be min/max for power-of-2 values. For example, `8:256`, would sweep `8,16,32,64,128,256`
+
+#### `--sweep-list` - A comma-separated list of values that stimulus will be swept over
+
+### CLI Examples
+```bash
+genai-perf analyze -m <model> --sweep-type concurrency --sweep-range 1:256
+```
+This will sweep over concurrencies of 1,2,4,8,16,32,64,128, and 256
+
+```bash
+genai-perf analyze -m <model> --sweep-type request_rate --sweep-range 100:500:50
+```
+This will sweep over request rate at values of 100,150,...450,500
+```bash
+genai-perf analyze -m <model> --sweep-type input_sequence_length --sweep-list 100,150,200,400
+```
+This will sweep over ISL for values of 100,150,200 and 400
+
+## Artifact Directories and Summary CSV Report
+
+As when running `profile`, an artifact directory will be created for each scenario profiled. The name of the artifact directory is:
+`<model_name>-<service-kind>-<backend>-<sweep_type><sweep_value>`
+
+### Example
+For this command:
+```bash
+genai-perf analyze -m gpt2 --service-kind triton --backend vllm --sweep-type num_dataset_entries --sweep-range 100:200:50
+```
+
+The following artifact directories would be created:
+```bash
+artifacts/gpt2-triton-vllm-num_dataset_entries100/
+artifacts/gpt2-triton-vllm-num_dataset_entries150/
+artifacts/gpt2-triton-vllm-num_dataset_entries200/
+```
+
+Each artifact directory contains the `inputs.json`, `profile_export.json`, and `profile_export_genai_perf.csv` just like it would if you ran `profile` individually for each scenario.
+
+## Summary Report CSV
+In the CWD a summary report CSV (`analyze_export_genai_perf.csv`) is created. In the first table, each row is a different scenario profiled while the columns show p99 perf metrics. In the second table, the rows are again scenario's profiled, while the columns are p99 GPU telemetry metrics.
+
+### Example Summary Report CSV
+```
+Config Name,Concurrency,ISL,Num Dataset Entries,p99 Time To First Token (ms),p99 Inter Token Latency (ms),p99 Request Latency (ms),p99 Output Sequence Length (tokens),Avg. Output Token Throughput (tokens/sec),Request Throughput (requests/sec)
+gpt2_run_config_2,1,201,200,33.54,7.16,779.75,132.10,149.63,1.32
+gpt2_run_config_1,1,201,150,33.13,7.29,778.62,126.16,147.93,1.32
+gpt2_run_config_0,1,201,100,82.02,7.53,879.20,124.55,145.93,1.30
+
+Config Name,GPU,p99 GPU Power Usage (W),p99 GPU Energy Consumption (MJ),p99 GPU Utilization (%),p99 GPU Memory Used (GB),Avg. GPU Power Limit (W),Avg. GPU Total Memory (GB)
```

Comment:
Historically, we have had users complain that our csv files are delivered like this.
They don't technically abide by the csv format so they break when trying to use standard libraries to import.
Would it be easy to make this two different csv files? OR was this intentionally designed this way and I dont know the reason?

Reply:
Intentionally designed this way to match the CSV created in the artifact directory.

Reply:
I worry this is going to make it hard for the TMEs to parse the csv data.
It may need to iterate...but they should also just use the checkpoint

Reply:
It's easy enough to change - I have no problem modifying this to match whatever the TMEs want it to look like.